### ISSUE TITLE
Error: Prune: last wallet synchronisation goes beyond pruned data

### ISSUE THREAD

**Issue by unsystemizer** [bitcoin/bitcoin#11360]
On startup, I get "Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)"

As noted below, this message seems wrong as I was able to restart w/o issues - didn't reindex or download the blockchain again.

I am not asking for help with this. I just wanted to record this incident because #7494 was closed as fixed, but I ran into the same problem with 0.15.

### Can you reliably reproduce the issue?
No.

#### If so, please list the steps to reproduce below:
1. Run 0.14.2 in pruned mode
2. Stop/start/stop 0.14.2 several times
3. Upgrade to 0.15
3. Start 0.15

I am *not* sure how I might reproduce this, but it's very much like this issue here:
https://github.com/bitcoin/bitcoin/issues/7494#issue-132587553

I *might* have had a power failure or rebooted the host without shutting down this VM, I don't remember now. Today is Sep 18; I see in .bitcoin/chainstate that it seems Sep 10 was the last time I ran, and before that in August. 
It's possible that on Sep 10 I shut down the VM before it finished syncing which is one of possible reasons for this problem, mentioned in issue #7494. 

### What version of bitcoin-core are you using?
v0.15.0 (official binary)

### Machine specs:
- OS: Ubuntu 16.04 LTS
- CPU: i7
- RAM: 16 G
- Disk size: 200 G
- Disk Type (HD/SDD): SSD

### Any extra information that might be useful in the debugging process.

I just restarted. I did delete .lock before that, but I don't think it mattered.

Next time I started, there was no sign of any trouble. It doesn't look like Bitcoin had to completely resync .
Wallet seems to be working OK (I don't have any funds in it, but getbalance and such doesn't give any errors; I never used wallet so it's empty). 

Fail log:

```
# tail -f .bitcoin/debug.log
2017-09-18 12:58:57 init message: Loading block index...
2017-09-18 12:58:57 Opening LevelDB in /root/.bitcoin/blocks/index
2017-09-18 12:58:57 Opened LevelDB successfully
2017-09-18 12:58:57 Using obfuscation key for /root/.bitcoin/blocks/index: 0000000000000000
2017-09-18 12:59:00 LoadBlockIndexDB: last block file = 994
2017-09-18 12:59:00 LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=58, size=46642763, heights=483555...484527, time=2017-09-05...2017-09-10)
2017-09-18 12:59:00 Checking all blk files are present...
2017-09-18 12:59:00 LoadBlockIndexDB(): Block files have previously been pruned
2017-09-18 12:59:00 LoadBlockIndexDB: transaction index disabled
2017-09-18 12:59:00 Opening LevelDB in /root/.bitcoin/chainstate
2017-09-18 12:59:02 Opened LevelDB successfully
2017-09-18 12:59:02 Using obfuscation key for /root/.bitcoin/chainstate: 268af1ce9bee0cb3
2017-09-18 12:59:02 Loaded best chain: hashBestChain=0000000000000000011c5c5d5158781964fa1c62c76436a7c048f3b60bf767ce height=484527 date=2017-09-10 15:00:52 progress=0.991703
2017-09-18 12:59:02 init message: Rewinding blocks...
2017-09-18 12:59:03 init message: Verifying blocks...
2017-09-18 12:59:03 Verifying last 6 blocks at level 3
2017-09-18 12:59:03 [0%]...[16%]...[33%]...[50%]...[66%]...[83%]...[99%]...[DONE].
2017-09-18 12:59:08 No coin database inconsistencies in last 7 blocks (8434 transactions)
2017-09-18 12:59:08  block index           10261ms
2017-09-18 12:59:08 init message: Loading wallet...
2017-09-18 12:59:08 nFileVersion = 140100
2017-09-18 12:59:08 Keys: 102 plaintext, 0 encrypted, 102 w/ metadata, 102 total
2017-09-18 12:59:08  wallet                    6ms
2017-09-18 12:59:08 keypool added 900 keys (0 internal), size=1000 (0 internal)
2017-09-18 12:59:08 Error: Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)
2017-09-18 12:59:08 scheduler thread interrupt
2017-09-18 12:59:08 Shutdown: In progress...
2017-09-18 12:59:08 Shutdown: done
2017-09-18 12:59:08 CDBEnv::EnvShutdown: Error 22 shutting down database environment: Invalid argument
^C
```

After that I checked db.log:

```
# cat db.log
Database handles still open at environment close
Open database handle: wallet.dat/main
```

**Comment by MarcoFalke** [comment]
Is this still an issue with a recent version of Bitcoin Core? If yes, what are the steps to reproduce?

**Comment by Philomorph** [comment]
I just got this error with the latest version of core (0.21.0).
I haven't used BitCoin since 2017. I installed core last night and started downloading the blockchain overnight. I left the prune setting to the default 2Gb.
When I woke up it was up to November 2017, and the date on my wallet file was October 2017, so I figured I'd load it up and see if it had anything in it.
I closed the core client completely, copied the wallet file to the wallets folder, and restarted core.
It gave this error and then closed when I hit OK.
I started core up again and it's acting normally, continuing to download. The wallet screen says balance info is out of date, but that's the only indication it's not caught up.
I have another half a day probably for it to catch up - not sure what will happen when it does, but figured I'd report the error.
